compiler-plugin/src/main/java/io/ballerina/lib/np/compilerplugin/PromptAsCodeCodeModificationTask.java

```diff
@@ -362,6 +367,69 @@ public static MappingConstructorExpressionNode getAnnotationExpression(String js
         return (MappingConstructorExpressionNode) NodeParser.parseExpression(jsonSchema);
     }
 
+    private static boolean containsBallerinaxNpImport(NodeList<ImportDeclarationNode> imports) {
+        for (ImportDeclarationNode importDeclarationNode : imports) {
+            Optional<ImportOrgNameNode> importOrgNameNode = importDeclarationNode.orgName();
+            if (importOrgNameNode.isPresent() && importOrgNameNode.get().orgName().text().equals(ORG_NAME)
+                    && importDeclarationNode.moduleName().get(0).text().equals(MODULE_NAME)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static NodeList<ImportDeclarationNode> updateImports(ModulePartNode modulePartNode) {
+        NodeList<ImportDeclarationNode> imports = modulePartNode.imports();
+        NodeList<ModuleMemberDeclarationNode> members = modulePartNode.members();
+        if (containsBallerinaxNpImport(imports)) {
+            return imports;
+        }
+
+        for (ModuleMemberDeclarationNode memberNode : members) {
+            if (memberNode.kind() != SyntaxKind.TYPE_DEFINITION) {
+                continue;
+            }
+
+            TypeDefinitionNode typeDefinitionNode = (TypeDefinitionNode) memberNode;
+            NodeList<AnnotationNode> annotations = getMetadataNode(typeDefinitionNode).annotations();
+            for (AnnotationNode annotation: annotations) {
+                if (isNPSchemaAnnotationAvailable(annotation)) {
+                    return imports.add(createImportDeclarationNodeForModule());
+                }
+            }
+        }
+        return imports;
+    }
+
+    public static boolean isNPSchemaAnnotationAvailable(AnnotationNode annotationNode) {
+        if (annotationNode.annotReference() instanceof SimpleNameReferenceNode refNode) {
+            return refNode.name().text().equals(MODULE_NAME + ":" + SCHEMA_ANNOTATION_IDENTIFIER);
+        }
+        return false;
+    }
+
+    public static ImportDeclarationNode createImportDeclarationNodeForModule() {
```

Comment:
We can do this using NodeFactory easily. But anyway NodeFactory is also ok

Reply:
NodeParser, right? @chiranSachintha let's update this - e.g., https://github.com/MaryamZi/module-ballerinax-np/blob/compile-time-prompt-as-code/compiler-plugin/src/main/java/io/ballerina/lib/np/compilerplugin/CompileTimePromptAsCodeCodeModificationTask.java#L327

Reply:
Changed